server/src/main/java/org/elasticsearch/index/engine/DeleteVersionValue.java

```diff
@@ -12,13 +12,13 @@
 
 /** Holds a deleted version, which just adds a timestamp to {@link VersionValue} so we know when we can expire the deletion. */
 
-final class DeleteVersionValue extends VersionValue {
+public final class DeleteVersionValue extends VersionValue {
```

Comment:
I've reverted as many changes to visibility (needed for testing) as I could. There are a few left which I think are necessary.